src/components/WalletModal/TokensTab/TokensTab.tsx

```diff
@@ -1,71 +1,73 @@
-        <TokenIcon name={name ?? 'empty'} />
-      </Col>
-      <Col span={18} style={{ paddingLeft: 10 }}>
-        <Row>
-          <Typography.Text strong style={{ fontSize: '16px' }}>
-            {name}
-          </Typography.Text>
-        </Row>
-        <Row align="bottom">
-          <Typography.Text style={{ fontSize: '10px' }} className="token-name">
-            {name}
-          </Typography.Text>
-        </Row>
-      </Col>
-      <Col span={4}>
-        <Typography.Text strong style={{ fontSize: '16px', float: 'right' }}>
-          {balance}
-        </Typography.Text>
-      </Col>
-    </Row>
-  </Box>
-);
-
-const TokensTab: React.FC = () => {
-  const [availableWalletAssets, setAvailableWalletAssets] = useState<
-    AssetAmount[] | undefined
-  >();
+interface TokenViewProps {
+  token: string[];
+}
 
-  useEffect(() => {
-    listWalletAssets().then(setAvailableWalletAssets);
-  }, []);
+const TokenView: React.FC<TokenViewProps> = ({ token }) => {
+  return (
+    <Flex alignItems="center">
+      <Flex.Item marginRight={1}>
+        <Typography.Text strong>{getShortAddress(token[0])}</Typography.Text>
+      </Flex.Item>
+    </Flex>
+  );
+};
 
+const TokenListItem: React.FC<TokenListItemProps> = ({ token }) => {
   return (
-    <Row topGutter={1}>
-      {availableWalletAssets &&
-        availableWalletAssets.map((token, key) => (
-          <Col span={24} key={key}>
-            <TokenListItem
-              name={token.asset.name}
-              balance={Number(
-                renderFractions(token.amount, token.asset.decimals),
-              )}
-            />
-          </Col>
-        ))}
-    </Row>
+    <Box padding={[2, 0]} transparent>
+      <Flex alignItems="center" className="tokens-tab">
+        <Flex.Item style={{ width: '45%', display: 'inline-flex' }}>
+          <Flex>
+            <Flex.Item marginRight={1} style={{ position: 'relative' }}>
+              <TokenIcon name={token[0]} className="tokens-tab__icon" />
+            </Flex.Item>
+            <Flex.Item marginLeft={7}>
+              <TokenView token={token} />
+              <Typography.Text className="tokens-tab__text">
+                {token[0]}
+              </Typography.Text>
+            </Flex.Item>
+          </Flex>
+        </Flex.Item>
+        <Flex.Item grow>
+          <Flex justify="flex-end">
+            <Typography.Text strong className="tokens-tab__balance">
+              {token[2]}
+            </Typography.Text>
+          </Flex>
+        </Flex.Item>
+      </Flex>
+    </Box>
   );
 };
 
-export { TokensTab };
+interface TokenListViewProps {
+  tokenList: string[][];
+}
+
+export const TokensTab: React.FC<TokenListViewProps> = ({ tokenList }) => {
+  return (
+    <Box transparent>
+      <Flex flexDirection="col">
+        <Flex.Item>
+          <Box transparent maxHeight={250} overflow>
+            {tokenList.map((item, index) => (
+              <TokenListItem key={index} token={item} />
```

Comment:
maybe we can use some unique item field for key or hash

Reply:
Get it. Will import TokenListItem and use unique item field.